### PR TITLE
refactor: remove concat type from concat model

### PR DESCRIPTION
**Pull request by domoritz**
Fixes https://github.com/vega/vega-lite/issues/6023

We now consistently treat a concat spec with 1 column as a vconcat.

### CODE REVIEW

File: src/compile/model.ts [hunk 1]
```diff
@@ -209,7 +209,7 @@ export abstract class Model {
 
     this.description = spec.description;
     this.transforms = normalizeTransform(spec.transform ?? []);
-    this.layout = isUnitSpec(spec) || isLayerSpec(spec) ? {} : extractCompositionLayout(spec, type, config);
+    this.layout = type === 'layer' || type === 'unit' ? {} : extractCompositionLayout(spec, type, config);
```
Comment (kanitw) [inline src/compile/model.ts]: why is this related?

File: src/compile/layoutsize/parse.ts [hunk 2]
```diff
@@ -19,16 +19,12 @@ export function parseLayerLayoutSize(model: Model) {
...
-  hconcat: 'height'
-};
-
 export function parseConcatLayoutSize(model: ConcatModel) {
   parseChildrenLayoutSize(model);
   const layoutSizeCmpt = model.component.layoutSize;
 
-  const sizeTypeToMerge = SIZE_TYPE_TO_MERGE[model.concatType];
+  const columns = model.layout.columns;
+  const sizeTypeToMerge = columns === 1 ? 'width' : columns === undefined ? 'height' : undefined;
```
Comment (kanitw) [inline src/compile/layoutsize/parse.ts]: change `concatType` to a getter (which determines type based on column)? 

So here we can make the code a bit more readable?

File: src/compile/model.ts (hunk 1 above)
Comment (domoritz) [reply src/compile/model.ts]: extractCompositionLayout now has a type for column while it was any before.

File: src/compile/layoutsize/parse.ts (hunk 2 above)
Comment (domoritz) [reply src/compile/layoutsize/parse.ts]: Would that actually be more readable? I feel this code is very clear about what it does and you don't need to look at another function. I can make this a case statement, though.

Comment (kanitw) [reply src/compile/layoutsize/parse.ts]: still irrelevant to this PR, but sure I can ignore it.

File: src/compile/model.ts (hunk 1 above)
Comment (kanitw) [reply src/compile/model.ts]: still irrelevant to this PR, but sure I can ignore it.

Comment (domoritz) [reply src/compile/model.ts]: Okay
